### ISSUE TITLE
Wrong formatting applied to fragment of code with #if clause

### ISSUE THREAD

**Issue by knocte** [fsprojects/fantomas#2361]
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAdAdgWwQVwBtEwAFAJwEtsKAXCgN0QGcwBeLCCGgTwAdiAQgVjQA1gAlEFAOYALGmw6dlAHzBCREqXIWwAZmDwVMNAMwAmMFiUQAhpnhYAxBQMBCAHKwAIgFUmAIJMAMo0ZHjQNEw2ymAA2gA8oeGRAHwAus6IDq4xyhpikjLyAPJ6ekyINACMAGyKmLFNYGoFWsU0ZRVVdWD6hsY1tdaYkAA04FCVNHSY0kyo8ZMQoE2QACQ8-IuQg%2BPLnBv0tgR4yGhxB8oADBPNUACSDtk0wRQAXsh3zQAsV2DpA4AX2%2BK3%2BGy25yge1Bh3Wx1OUMu92qsOUkAAsrYAB4AGWMiFx2WkNFk%2BxR5mu-0ByhBB1WsQhfChkGy8DK%2BMwiFC3CI5MZ8JOZ0WyOa5jRhwAog4OQT%2BWsCHpINTgbCGejNsydgAjWCwAjyuEI4UXf4QUwSiCQJ6VMg0ABixhOHkQAHcCHLLWAwmcVbS1eDNds0JBdfrDVbBYiRWawD8vZBgrxbNBEAJEHpYGRECRbGRbNgqogyBHOD7EH7OHTlOq4ZCdXqDRKjkKkbGAKwJpMptMZrOE2Cu4vQWyVJ70ES2OiwUZe8uViDVzi1yP1kNhpuB41t%2B61LvJ1PpzPZ3y8fhkEdjzATkfT2exvQnSoLsBLsFrIMsjellvR033AB2fceyPfsAGECFHJgwJnJgfUiLNSwgR8CGfJoaSrAMPzXKBv2bKMTSWe4AA5gMPPtswxRBsG1YskLAFC0NiDDFywgUcNDRsfwIndmgATjI3tj0QGDhHve5GIrdDVXpQMOLwrdWxjFFbljRMD0QAI9BoYsYOwbBbHo%2BdpP9WTsK1dcuPw7dlOaapUTU7tyOE4JqIoOAxPoySXzfMAVygT8G3DaylP-OzxUcjStJ0shXOoDyZyM8IpOYmSazkizcKsxS-yIuyLUinsAjIWA8AcbxEA9agYqS30TMwsz2My3YTG4mywqaap4zUrFsQePQABVZGyYJZCzGgAHUKHgUl6KperWMajUOJhHLCNFTrOx6nF%2BqG7JJVQ7kxrtKaZrJL0fnm1LTPS8zg2hVqQtyjbYjqBNeqePQKGxEpzynLNJWxXhsyYJgKESr12yu5QWNfNjlua1aP3avLOqA7bsQAJUQOAyHgU7Zou6HOFh3z-KZe6WpoNrQtR17SIx7HcfgDw8Bo4sygeHTsAWL1qh8%2BG62a7BCDoD0uXtLMDJmYsBq1J71tjaoBLUpms3gDFRYocXEElshpZqhNoFkPMUxigB9V1ptmgWlqFymkYFFGXuUSl3pxYr824EoyHxOCCfO2NLtt26modx61t4ppzAc%2B5MQ9sgvZ9v2aFZ9myE57neaVkPlwyymRYIMWCT1g3ZflyPbOjiK4892xvd9ig4M1ovtZLqWp0NtTjdNyJi0t62yVz98w5ZR2NWd2NzAKuPeoANVbARjHgYxpAD%2BjiOJiBScF1dEYj5HaZdzhzG62ecXtMrInBzAl5yOZ16J4e-PzseD6do%2Bp628-sW8WAaAAOJVEBsDZgYMZyPw7FvAEaU853S-NlQ%2Bz0p57h6lrHWbQBD5jEFUJgJRMDBALCJfUbNxLNG8gtOGds96UwUkgxW9xzDozji6d0BJ0w0CHNkOW-AKpfUwLQG%2BAQHBUXTtnCST4Uow1gSPBGtDEEf2QYwhmtcPTSEwJfTA18ZxvHUVOPA2YBqwCeEgEwU4b5eUkc-cmgVLLBUrh1WI5gVaqJipgcxjBcQzmkKI2iZA%2BHGEEbBSxh1rGvyCpuehUdYimFUrPLWnJCQFm1PAWwEFYBgzmKwnWISmLSJunA0eESaZKOaKYWOzRIAgOLFQF4JwADSiBEC8BMS8J4WD7DG1yVIkmMiX7wOKQraJygLAJiEPYUQiTAglTKizZgOkNboIJCAkG4DMDiOaMZa6DVQ5yIQfYqJVcYkz0qQIPMoFszeCbtASg1B3ELN8AImcFVoCQXzHebpYSBl2MiYohhZSz6VOqbcupBBkilTguEXgPI%2BRegodsxauz7Zv2pkMo5Izv6VMac03qac-FlHGZgSZBINmdWuNAne1CAryQURPT%2B9xTCoLjskdyNAMQIC%2BA%2BKxlCgSTHSN8XYTB7Rg0WN5IEQA)

#### Code

```fsharp

module Primitives =
    type BlockHeight =
        | BlockHeight of uint32 

    and
#if !NoDUsAsStructs
        [<Struct>]
#endif
        BlockHeightOffset16 =
            | BlockHeightOffset16 of uint16


```
            

#### Result

```fsharp
module Primitives =
    type BlockHeight = BlockHeight of uint32

#if !NoDUsAsStructs
    and [<Struct>] BlockHeightOffset16
#endif
    and BlockHeightOffset16 = BlockHeightOffset16 of uint16

```
            
#### Problem description

If you compile the result with NoDUsAsStructs defined, it works; but otherwise it doesn't because this code doesn't compile:
```
    and [<Struct>] BlockHeightOffset16
    and BlockHeightOffset16 = BlockHeightOffset16 of uint16
```

#### Extra information

- [x] The formatted result breaks my code.
- [ ] The formatted result gives compiler warnings.
- [x] I or my company would be willing to help fix this.

#### Options

Fantomas master branch at 2022-07-11T16:44:40Z - 98a7449918c2167c0478a151ec89bbde826a74c0

Default Fantomas configuration

<sub>Did you know that you can ignore files when formatting from fantomas-tool or the FAKE targets by using a [.fantomasignore file](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#ignore-files-fantomasignore)?</sub>

**Comment by knocte** [comment]
Expected result (I guess):

```
module Primitives =
    type BlockHeight = BlockHeight of uint32 

    and
#if !NoDUsAsStructs
        [<Struct>]
#endif
        BlockHeightOffset16 = BlockHeightOffset16 of uint16
```

**Comment by nojaf** [comment]
Fix available in [v5.2.0-beta-001](https://www.nuget.org/packages/fantomas/5.2.0-beta-001)
